include/signer/signer.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <future>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "data/data.hpp"
+
+namespace nostr
+{
+namespace signer
+{
+enum class Encryption
+{
+    NIP04,
+    NIP44
+};
+
+/**
+ * @brief An interface for Nostr event signing that implements NIP-46.
+ */
+class ISigner
```

Comment:
Should we consider renaming since abstract signers will be doing all forms of nostr private operations? It's almost more of a credential store with some extra features.

Reply:
Within the Nostr domain, I think "signer" is understood to be the thing responsible for handling private operations, so it should be understandable.  Do you have suggestions for an alternate name, though?

Reply:
Yeah, maybe I'm being pedantic, but there is no standard and I think it's silly were calling it a "signer" when it does much more than sign, which I'd argue is only a colloquial term we've been using.

Reply:
What does it do, beyond just signing?  As far as I see it, everything the signer does is signing or signing-adjacent.